### PR TITLE
Add kubectl datadog plugin doc

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

Add documentation about the kubeclt plugin

### Motivation

the kubectl plugin is now part of the release process.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: docs/kubectl-plugin.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# kubectl datadog plugin
+
+## install the plugin with krew
+
+The Datatadog operator comes with a kubectl plugin that provides a set of helper went you want to get some inside components information.
+
+The installation is done with the [plugin manager krew](https://krew.sigs.k8s.io/)
```
Comment (xornivore) [inline docs/kubectl-plugin.md:7]:
```suggestion
To install please use the [krew plugin manager](https://krew.sigs.k8s.io/)
```

File: docs/kubectl-plugin.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+The krew plugin manifest url can be found on the project [release page](https://github.com/DataDog/datadog-operator/releases). Each release has an artifact: `datadog-plugin.yaml` manifest file.
```
Comment (xornivore) [inline docs/kubectl-plugin.md:9]:
```suggestion
The krew plugin manifest url can be found on the project [release page](https://github.com/DataDog/datadog-operator/releases). Each release has its own `datadog-plugin.yaml` manifest file.
```

File: docs/kubectl-plugin.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+Installed plugin: datadog
+\
+ | Use this plugin:
+ | 	kubectl datadog
+ | Documentation:
+ | 	https://github.com/DataDog/datadog-operator
+/
+```
+
+## available commands
```
Comment (xornivore) [inline docs/kubectl-plugin.md:23]:
```suggestion
## Available commands
```

File: docs/kubectl-plugin.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  agent
+  clusteragent
+  flare        Collect a Datadog's Operator flare and send it to Datadog
+  get          Get DatadogAgent deployment(s)
+  help         Help about any command
+  validate
+
+```
+
+### agent sub-commands
```
Comment (xornivore) [inline docs/kubectl-plugin.md:40]:
```suggestion
### Agent sub-commands
```

File: docs/kubectl-plugin.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  datadog agent [command]
+
+Available Commands:
+  check       Find check errors
+  find        Find datadog agent pod monitoring a given pod
+  upgrade     Upgrade the Datadog Agent version
+
+```
+
+### clusteragent sub-commands
```
Comment (xornivore) [inline docs/kubectl-plugin.md:54]:
```suggestion
### Cluster Agent sub-commands
```

File: docs/kubectl-plugin.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# kubectl datadog plugin
```
Comment (xornivore) [inline docs/kubectl-plugin.md:1]:
```suggestion
# Datadog Operator kubectl plugin
```

File: docs/kubectl-plugin.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+```console
+$ kubectl datadog clusteragent --help
+Usage:
+  datadog clusteragent [command]
+
+Available Commands:
+  leader      Get Datadog Cluster Agent leader
+  upgrade     Upgrade the Datadog Cluster Agent version
+```
```
Comment (ahmed-mez) [inline docs/kubectl-plugin.md:64]: we can add a section for the Validate command

Comment (ahmed-mez) [reply docs/kubectl-plugin.md:64]:
```
./kubectl-datadog validate ad                                   
Usage:
  datadog validate ad [command]

Available Commands:
  pod         Validate the autodiscovery annotations for a pod
  service     Validate the autodiscovery annotations for a service
```

Comment (clamoriniere) [reply docs/kubectl-plugin.md:64]: done

File: docs/kubectl-plugin.md [hunk 8]
```diff
@@ -62,3 +62,15 @@ Available Commands:
   leader      Get Datadog Cluster Agent leader
   upgrade     Upgrade the Datadog Cluster Agent version
 ```
+
+### Validate sub-commands
+
+```console
+$ kubectl datadog validate --help
```
Comment (ahmed-mez) [inline docs/kubectl-plugin.md:69]:
```suggestion
$ kubectl datadog validate ad --help
```

Comment (clamoriniere) [reply docs/kubectl-plugin.md]: done

File: docs/kubectl-plugin.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
+# Datadog Operator kubectl plugin
+
+## Install the plugin with krew
+
+The Datatadog Operator comes with a kubectl plugin providing a set of helper utilities giving visibility into certain internal components which some users may find useful.
```
Comment (sarina-dd) [inline docs/kubectl-plugin.md:5]:
```suggestion
The Datadog Operator comes with a kubectl plugin providing a set of helper utilities giving visibility into certain internal components which some users may find useful.
```

File: docs/kubectl-plugin.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+To install please use the [krew plugin manager](https://krew.sigs.k8s.io/)
```
Comment (sarina-dd) [inline docs/kubectl-plugin.md:7]:
```suggestion
To install, use the [krew plugin manager](https://krew.sigs.k8s.io/).
```

**Pull request merged**